### DISCUSSION TITLE
Will it work for Angular 7?

### DISCUSSION THREAD

**Discussion by yankycranky** [panva/jose#35] [Q&A]
I have Angular 7 based web-app, can i use this library in it. I tried integrating it, however i am getting "crypto"' has no exported member 'KeyObject' and other three.. i have nodeJs 12.7.0 installed

**Comment by panva** [comment]
No, you [cannot](https://github.com/panva/jose#how-do-i-use-it-outside-of-nodejs) use @panva/jose outside of Node.js or (recently released) Electron JS runtimes.

No amount of transpiling and polyfills can make up for the design differences between node's `crypto` module and browser's Web Crypto API

**Comment by MuhammadBilalYar** [comment]
I am also facing these errors after installing OpenId-Connect@V3.7 in my ExpressJs app

node_modules/jose/types/index.d.ts:6:10 - error TS2305: Module '"crypto"' has no exported member 'KeyObject'.

6 import { KeyObject, PrivateKeyInput, PublicKeyInput } from 'crypto';
           ~~~~~~~~~

node_modules/jose/types/index.d.ts:6:21 - error TS2305: Module '"crypto"' has no exported member 'PrivateKeyInput'.

6 import { KeyObject, PrivateKeyInput, PublicKeyInput } from 'crypto';
                      ~~~~~~~~~~~~~~~

node_modules/jose/types/index.d.ts:6:38 - error TS2305: Module '"crypto"' has no exported member 'PublicKeyInput'.

6 import { KeyObject, PrivateKeyInput, PublicKeyInput } from 'crypto';

is this known issue? any work around? 

**SOLUTION:
Update @type/node version to V12**

**Comment by panva** [comment]
@MuhammadBilalYar could it be your `@types/node` is not the latest (v12)? the types borrow those definitions from it.

**Comment by MuhammadBilalYar** [comment]
@panva My bad, after updating version issue resolved. Thanks

**Comment by panva** [comment]
A quick note, i've release a new major (v3.x) which is now a universal module with 0 dependencies and now works the same in the browser (using webpack 5 with no issues) as well as node.
